data_generation/generate.js

```diff
@@ -0,0 +1,56 @@
+const fs = require('graceful-fs');
+const path = require('path');
+
+let currentId = 1;
+let num;
+const names = ['Voyager', 'Ghost', 'Slate', 'Eclipse', 'Chrono', 'Myst', 'Argo', 'Vice', 'Dune', 'Calypso', 'Rogue', 'Avalon', 'Denali', 'Mariner', 'Maverick', 'Abyss', 'Atlas'];
+const start = new Date().getTime();
+
+const randomPhotoNum = (id) => {
+  const low = 100;
+  const high = 160;
+  if (id >= low && id <= high) {
+    num = id;
+  } else {
+    num = Math.floor(Math.random() * (high - low + 1)) + low;
+  }
+  return num;
+};
+
+const randomArrayElement = (array) => {
+  const idx = Math.floor(Math.random() * array.length);
+  return array[idx];
+};
+
+const csvGenerate = (id) => {
+  const row = `${id},${randomArrayElement(names)},https://s3-us-west-1.amazonaws.com/restwatch-product-photos/${randomPhotoNum(id)}_front.jpg,https://s3-us-west-1.amazonaws.com/restwatch-product-photos/${num}_side.jpg,https://s3-us-west-1.amazonaws.com/restwatch-product-photos/${num}_back.jpg,https://s3-us-west-1.amazonaws.com/restwatch-product-photos/box.jpg,https://s3-us-west-1.amazonaws.com/restwatch-product-photos/${num}_style.jpg\n`;
+  const csvOutput = id === 1
+    ? `id,name,frontImg,sideImg,backImg,box,styleImg\n${row}`
+    : `${row}`;
+  currentId += 1;
+  return csvOutput;
+};
+
+const generateData = (entries) => {
+  const write = fs.createWriteStream(path.join(__dirname, '/data/photo_data.csv'));
+
+  const writeData = (entriesDone) => {
+    if (currentId >= entries) {
+      const end = new Date().getTime();
+      console.log('Write time: ', (end - start) / 1000, ' seconds');
+      return;
+    }
+    const ok = write.write(csvGenerate(currentId));
```

Comment:
On the next refactor can you think about how your variables can be better named? 'write.write', etc.